### PR TITLE
Report missing book front end

### LINKED ISSUE #174: Report missing book
**As a** Kamu user
**When** I see a book in Kamu but I can't find the physical copy of it
**I want to** report that the book is missing
**So that** the library admins can know that it's missing and update it or remove it from the library

It's pretty common when having a library that sometimes a book goes missing from that location - maybe because a user borrowed the book and forgot to borrow on Kamu, someone misplaced it or something else. Usually people send an email asking where a book is, but when it cannot be found, it often is left on Kamu. And so the frustration repeats for the next person that looks for it.

The idea of this feature request is to have a report option on the book. Here are some possible use cases:
* Display a warning on the book itself, to let interested users that it was previously reported missing;
* Display a list of reports on the admin page, so that library admins can check follow up on them and update or remove the books
* Send an automated notification to the library users asking if someone knows where the book is
etc.

### PR DESCRIPTION
**Pull request by erikaya**
Closes #174 

Creates a button inside the book details popup to report book as missing or mark book as found

Evidences:

report-missing

report-found

report-missing-dark-mode

report-found-dark-mode

---
This change is [Reviewable](https://reviewable.io/reviews/ayr-ton/kamu/186)

### CODE REVIEW

**Comment by erikaya** [comment]
In the future I want to work on the current tests that we have. My idea is to purpose an approach that can be more focused on behavior other than focusing on implementation details.

File: assets/src/components/books/detail/BookDetail.test.jsx [hunk 1]
```diff
@@ -34,7 +38,7 @@ describe('Book Detail', () => {
     const bookDetail = shallowBookDetail({ ...testDefaultProps });
     expect(
       bookDetail.find('.modal-book__details div.modal-book__title').text(),
-    ).toEqual('Test Driven Development');
+    ).toEqual('Test Driven Development<BookActionButton />');
```
Comment (erikaya) [inline assets/src/components/books/detail/BookDetail.test.jsx:41]: Not sure if is the right way to do it, just did it to fix the tests :|

Comment (macecchi) [reply assets/src/components/books/detail/BookDetail.test.jsx:41]: It's an issue with some of the older tests that still use Enzyme's shallow rendering. We should get rid of it to use react-testing-library's mount instead, but that will probably require some refactoring.

**Review by macecchi — APPROVED** [review]
👏 👏 👏

**Pull request merged**